src/services/dashboards/dashboards.svc.coffee

```diff
@@ -78,6 +83,19 @@ angular
     # Loaders and setters
     #====================================
 
+    # Method used for reloading an already loaded dashboard, will reload it properly
+    # while also triggering the dhbLoader spinner.
+    @reload = (force=false) ->
+      deferred = $q.defer()
+      _self.triggerDhbLoader(true)
+      _self.load(force).then(
+        -> deferred.resolve()
+        -> deferred.reject()
+      ).finally(->
+        _self.triggerDhbLoader(false)
+      )
+      return deferred.promise
+
```

Comment:
@cesar-tonnoir @alexnoox I tested it guys, chaining a promise with finally does not work, as it the finally callback is called before the clients (invoker) success or error handler is called.